### PR TITLE
PE-928: validate drive name when creating new drives

### PR DESCRIPTION
**Pull request by karlprieb**
This PR is heavily based on @matibat work on rename files, folders and drives feature.

- improves ArFS entity name validators tests making them parameterized
- add drive name validation to `createPublicDrive` and `createPrivateDrive` integration tests
- add drive name validation to drive creation methods

### CODE REVIEW

**Review by jdaev — COMMENTED** [review]
LGTM

**Review by jdaev — APPROVED** [review]

**Review by fedellen — APPROVED** [review]
Looks good!

**Review by matibat — APPROVED** [review]
I like how the unit tests for the validators got consolidated :clap: 
LGTM!

File: src/ardrive.ts [hunk 1]
```diff
@@ -1204,14 +1205,20 @@ export class ArDrive extends ArDriveAnonymous {
 	}
 
 	public async createPublicDrive(params: CreatePublicDriveParams): Promise<ArFSResult> {
+		const { driveName } = params;
+
+		assertValidArFSDriveName(driveName);
+
 		return this.createDrive(getPublicCreateDriveEstimationPrototypes(params), (rewardSettings) =>
-			this.arFsDao.createPublicDrive({ driveName: params.driveName, rewardSettings })
+			this.arFsDao.createPublicDrive({ driveName: driveName, rewardSettings })
```
Comment (arielmelendez) [inline src/ardrive.ts:1213]: Nit: you can removed the `driveName:` param name now that you unpacked `driveName` from `params`

**Pull request merged**